adminSiteServer/apiRouter.ts

```diff
@@ -2543,4 +2545,118 @@ deleteRouteWithRWTransaction(
     }
 )
 
+// Get an ArchieML output of all the work produced by an author. This includes
+// gdoc articles, gdoc modular/linear topic pages and wordpress modular topic
+// pages. Data insights are excluded. This is used to manually populate the
+// [.secondary] section of the {.research-and-writing} block of author pages
+// using the alternate template, which highlights topics rather than articles.
+getRouteWithROTransaction(apiRouter, "/all-work", async (req, res, trx) => {
+    type WordpressPageRecord = Record<
+        | "slug"
+        | "title"
+        | "subtitle"
+        | "type"
+        | "thumbnail"
+        | "authors"
+        | "publishedAt",
+        string
+    >
+    type GdocRecord = Pick<DbRawPostGdoc, "id" | "publishedAt"> &
+        Pick<OwidGdocContent, "title" | "type">
+
+    const author = req.query.author || "Max Roser"
+    const gdocs = await db.knexRaw<GdocRecord>(
+        trx,
+        `-- sql
+            SELECT id, content->>'$.title' as title, content->>'$.type', publishedAt
```

Comment:
We added [a virtual `type` column](https://github.com/owid/owid-grapher/blob/57003a169f1ac969ef11ee9bee74d0ec8b18d62c/db/migration/1712843782834-AddVirtualGdocsTypeColumn.ts#L9) that you can select from directly, which is why this query is working even though the records are actually of the shape `{ id: string, title: string, "content->>'$.type'": string, publishedAt: Date }` at the moment.

`GdocRecord` only needs `id` and `publishedAt`, and in the query for `wpModularTopicPages`, you could instead have `TRUE AS isWordpressPage` which you could use in your `isWordpressPage` type guard.

I don't think any of this really matters, of course 😛 just some technical correctness nitpicking.

Reply:
@ikesau thanks for the suggestions, there were indeed some leftovers from a previous attempt to use a UNION, which ran out of memory on sorting, as usual... 

That all makes sense to me, except the part about the virtual column preventing the query from failing (or maybe you were referring to some downstream code?). Indeed, `content->>$.type` as a column name was being passed but not used by the type guard. On the other hand, `type` (the virtual column) could have been used but wasn't being SELECTed. So this meant `type` was undefined for gdocs, which happened to be compatible with the type guard.

Maybe I missed your point?